src/cpp/r/RSexp.cpp

```diff
@@ -172,24 +196,58 @@ SEXP findVar(const std::string& name, const std::string& ns)
    
    return findVar(name, env);
 }
-  
+
 SEXP findFunction(const std::string& name, const std::string& ns) 
 {
+   r::sexp::Protect protect;
```

Comment:
It looks like there are only a couple sparse uses of `findFunction` currently: two in `SessionHelp.cpp` and one (the main use) in `RExec.cpp`:

r/RExec.cpp
320:   functionSEXP_ = sexp::findFunction(name, ns);

session/modules/SessionHelp.cpp
789:   return r::sexp::findFunction(".rs.handlerLookupError");
849:                      boost::bind(r::sexp::findFunction, "httpd", "tools"),

I'll double check both of those code paths to be sure this wouldn't affect how they run.